internal/backends/android-activity/lib.rs

```diff
@@ -340,6 +360,37 @@ impl AndroidWindowAdapter {
                     MotionAction::HoverEnter | MotionAction::HoverExit => InputStatus::Unhandled,
                     _ => InputStatus::Unhandled,
                 },
+                InputEvent::TextEvent(state) => {
+                    let runtime_window = i_slint_core::window::WindowInner::from_pub(&self.window);
+                    // remove the pre_edit
+                    let event = if let Some(r) = state.compose_region {
+                        i_slint_core::input::KeyEvent {
+                            event_type: i_slint_core::input::KeyEventType::UpdateComposition,
+                            text: i_slint_core::format!(
+                                "{}{}",
+                                &state.text[..r.start],
+                                &state.text[r.end..]
+                            ),
+                            preedit_text: state.text[r.start..r.end].into(),
+                            preedit_selection: Some(r.start as i32..r.end as i32),
```

Comment:
Shouldn't this be `Some(0..r.len() as i32)`? We define `preedit_selection` to be relative within the pre-edit text ("/// The selection within the preedit_text") and AFAICS that's also how we implement it in `Text` by making it absolute like this before rendering:

```rust
let selection_range =
                cursor_position + preedit_selection_start..cursor_position + preedit_selection_end;
```